packages/core/src/modules/mdoc/Mdoc.ts

```diff
@@ -0,0 +1,121 @@
+    }
+    if (Object.values(JwaSignatureAlgorithm).includes(algName as JwaSignatureAlgorithm)) {
+      return algName as JwaSignatureAlgorithm
+    }
+    throw new MdocError(`Cannot parse mdoc. The signature algorithm '${algName}' is not supported.`)
+  }
+
+  public get validityInfo() {
+    return this.issuerSignedDocument.issuerSigned.issuerAuth.decodedPayload.validityInfo
+  }
+
+  public get deviceSignedNamespaces(): MdocNameSpaces {
+    if (this.issuerSignedDocument instanceof DeviceSignedDocument === false) {
+      throw new MdocError(`Cannot get 'device-namespaces from a IssuerSignedDocument. Must be a DeviceSignedDocument.`)
+    }
+
+    return this.issuerSignedDocument.allDeviceSignedNamespaces
+  }
+
+  public get issuerSignedNamespaces(): MdocNameSpaces {
+    return this.issuerSignedDocument.allIssuerSignedNamespaces
+  }
+
+  public static async create(agentContext: AgentContext, options: MdocCreateOptions) {
+    const { docType, validityInfo, namespaces, holderPublicKey, issuerCertificate } = options
+    const mdocContext = getMdocContext(agentContext)
+
+    const holderPublicJwk = await getJwkFromKey(holderPublicKey)
+    const document = new Document(docType, mdocContext)
+      .useDigestAlgorithm('SHA-256')
+      .addValidityInfo(validityInfo)
+      .addDeviceKeyInfo({ deviceKey: holderPublicJwk.toJson() })
+
+    for (const [namespace, namespaceRecord] of Object.entries(namespaces)) {
+      document.addIssuerNameSpace(namespace, namespaceRecord)
+    }
+
+    const cert = X509Certificate.fromEncodedCertificate(issuerCertificate)
+    const issuerPrivateJwk = await getJwkFromKey(options.issuerKey ?? cert.publicKey)
+    const issuerSignedDocument = await document.sign(
+      {
+        issuerPrivateKey: issuerPrivateJwk.toJson(),
+        alg: issuerPrivateJwk.supportedSignatureAlgorithms[0] as 'ES256' | 'ES384' | 'ES512' | 'EdDSA',
+        issuerCertificate,
+        kid: cert.publicKey.fingerprint,
+      },
+      mdocContext
+    )
+
+    return new Mdoc(issuerSignedDocument)
+  }
+
+  public async verify(agentContext: AgentContext, options?: MdocVerifyOptions): Promise<boolean> {
+    const trustedCerts =
+      options?.trustedCertificates ?? agentContext.dependencyManager.resolve(X509ModuleConfig).trustedCertificates
+
+    if (!trustedCerts) {
+      throw new MdocError('No trusted certificates found. Cannot verify mdoc.')
+    }
+
+    const mdocContext = getMdocContext(agentContext)
+    try {
+      const verifier = new Verifier()
+      await verifier.verifyIssuerSignature(
+        {
+          trustedCertificates: trustedCerts.map((cert) => X509Certificate.fromEncodedCertificate(cert).rawCertificate),
+          issuerAuth: this.issuerSignedDocument.issuerSigned.issuerAuth,
+          disableCertificateChainValidation: false,
+          now: options?.now,
+        },
+        getMdocContext(agentContext)
+      )
+
+      await verifier.verifyData({ mdoc: this.issuerSignedDocument }, mdocContext)
```

Comment:
does it return a verification object or something? Would be good to return more than true/false, but also something we can add later

Reply:
doesnt return anything atm.